sdk/storage/Microsoft.Azure.WebJobs.Extensions.Storage.Common/src/Shared/Queues/QueueScaleMonitor.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Storage.Queues;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Storage.Common.Listeners
+{
+    /// <summary>
+    /// Used to retrieve metrics and make scale decisions for Queues.
+    /// </summary>
+    internal class QueueScaleMonitor : IScaleMonitor<QueueTriggerMetrics>
+    {
+        private const int NumberOfSamplesToConsider = 5;
+
+        private QueueClient _queue;
+        private ILogger _logger;
+        private ScaleMonitorDescriptor _scaleMonitorDescriptor;
+        private QueueMetricsProvider _queueMetricsProvider;
+
+        /// <summary>
+        /// Public constructor used to instantiate a QueueScaleMonitor for retrieving metrics and making scale decisions.
+        /// </summary>
+        /// <param name="functionId">The function name to make scale decisions for.</param>
+        /// <param name="queue">The queue client to poll metrics from.</param>
+        /// <param name="loggerFactory">Used to create an ILogger instance.</param>
+        public QueueScaleMonitor(string functionId, QueueClient queue, ILoggerFactory loggerFactory)
+        {
+            _queue = queue;
+            _logger = loggerFactory.CreateLogger<QueueListener>();
+            _scaleMonitorDescriptor = new ScaleMonitorDescriptor($"{functionId}-QueueTrigger-{_queue.Name}".ToLower(CultureInfo.InvariantCulture));
```

Comment:
use a new ctor from 3.0.36 which takes 2 params

Reply:
done, added `functionId` as the second parameter